lib/mix/lib/mix/tasks/format.ex

```diff
@@ -177,6 +177,18 @@ defmodule Mix.Tasks.Format do
     end
 
     {not_equivalent, not_formatted}
+
+  rescue
+    exception in SyntaxError ->
```

Comment:
Let's make this more general purpose. Something like:

```
rescue e ->
  stacktrace = System.stacktrace
  Mix.shell().error("mix format failed when parsing file: #{file}")
  reraise e, System.stacktrace
```

This way it works for all errors and we don't rewrite the exceptions code.

We can also say:

    Mix.shell().error("mix format failed when parsing file: #{file} (error shown below)")

What do you think?

Reply:
ok. This results in:

```
mix format failed when parsing file: lib/ui/templates/layout/no_user.html.slim
** (SyntaxError) nofile:5: syntax error before: http
    (elixir) lib/code.ex:329: Code.format_string!/2
    (mix) lib/mix/tasks/format.ex:148: Mix.Tasks.Format.format_file/4
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    (mix) lib/mix/tasks/format.ex:78: Mix.Tasks.Format.run/1
    (mix) lib/mix/task.ex:317: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:68: Mix.CLI.run_task/2
```

which I think it's good enough.

Reply:
Awesome, let's roll with it then! :D

Reply:
Done & PR is updated.